demos/object_detection_demo/jupyter_python/object_detection_demo.ipynb

```diff
@@ -0,0 +1,754 @@
+    "        display(HTML(f'<div class=\"alert alert-warning\" style=\"color:black\"><i><b>Warning: </b>{error_message}</div>'))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "national-plaintiff",
+   "metadata": {},
+   "source": [
+    "### Get model info\n",
+    "\n",
+    "The Info Dumper returns information for the Open Model Zoo models. It returns a list of dictionaries with the model name, description, framework, license url, precisions, task type, and the subdirectory for the downloaded model."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "naked-iraqi",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "info_command = os.path.join(open_model_zoo_path, \"tools\", \"downloader\", \"info_dumper.py\")\n",
+    "info_result = subprocess.run(\n",
+    "    [\n",
+    "        sys.executable,\n",
+    "        info_command,\n",
+    "        \"--list\",\n",
+    "        models_file,\n",
+    "    ],\n",
+    "    stdout=subprocess.PIPE,\n",
+    "    stderr=subprocess.PIPE,\n",
+    "    shell=False,\n",
+    "    universal_newlines=True,\n",
+    ")\n",
+    "info = json.loads(info_result.stdout)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "tropical-forest",
+   "metadata": {},
+   "source": [
+    "Make a list of models that will be shown as options. By default, only Intel models will be shown. Change this by uncommenting the second line to use all models. You need to make sure that the models are in IR format. This can be done by setting `CONVERT_MODELS` to `True` and running all cells. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "hungarian-opposition",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model_names = [model[\"name\"] for model in info if \"intel\" in model[\"subdirectory\"]]\n",
+    "# model_names = [model[\"name\"] for model in info]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "other-johns",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Show an example of the information that the Info Dumper returns\n",
+    "info[0]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "floral-longer",
+   "metadata": {},
+   "source": [
+    "The `models_file` file, by default \"models.lst\", lists models that are supported by this demo, sorted by architecture. The model names can contain wildcard. For example, `face-detection-????` means that the demo supports all models with a name that starts with `face-detection-` followed by four digits. \n",
+    "\n",
+    "We create a `model_architectures` dictionary that maps the model names given by the Info Dumper, to an architecture given by `models_file`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "engaging-personal",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model_architectures = {}\n",
+    "modellist = open(models_file).read().splitlines()\n",
```

Comment:
Use `with` to ensure the file is closed.

Reply:
The file itself is not assigned to a variable, only the result of read().splitlines(), so the file object itself should be discarded immediately and not remain in memory. No problem to change it to use with though.

Reply:
AFAIK, there is no guarantee that Python will perform garbage collection when you expect it to. If it works correctly, it's only because the current version of the implementation happens to behave that way. This is why you get a `ResourceWarning` when you don't close a file (if you enable this warning).